src/ts/component/page/main/date.tsx

```diff
@@ -114,6 +115,59 @@ const PageMainDate = observer(class PageMainDate extends React.Component<I.PageC
 		);
 	};
 
+	getFilters = (first: {y: number, m: number, d: number}, last: {y: number, m: number, d: number}): I.Filter[] => {
+		const { selectedRelation } = this.state;
+
+		if (!selectedRelation) {
+			return [];
+		};
+
+		const start = U.Date.timestamp(first.y, first.m, first.d, 0, 0, 0);
+		const end = U.Date.timestamp(last.y, last.m, last.d, 23, 59, 59);
+
+		return [
+
+			{
+				relationKey: selectedRelation,
+				condition: I.FilterCondition.GreaterOrEqual,
+				value: start,
+				quickOption: I.FilterQuickOption.ExactDate,
+				format: I.RelationType.Date,
+			},
+			{
+				relationKey: selectedRelation,
+				condition: I.FilterCondition.LessOrEqual,
+				value: end,
+				quickOption: I.FilterQuickOption.ExactDate,
+				format: I.RelationType.Date,
+			}
+		];
+	};
+
+	getDotMap = (data: {d: number, m: number, y: number}[], callback: (res: Map<string, boolean>) => void): void => {
+		const { rootId } = this.props;
+		const { selectedRelation } = this.state;
+
+		const first = data[0];
+		const last = data[data.length - 1];
+
+		U.Data.search({
+			filters: this.getFilters(first, last),
+			keys: [ selectedRelation ],
+		}, (message: any) => {
+			const res = new Map();
+			data.forEach(date => {
+				if (message.records.find(rec => {
+					const recDate = U.Date.getCalendarDateParam(rec[selectedRelation]);
+					return recDate.y == date.y && recDate.m == date.m && recDate.d == date.d;
```

Comment:
Parts of logical statements should be wrapped in parenthesis. Also I didn't find declaration for `date` variable.
```
(recDate.y == date.y) && (recDate.m == date.m) && (recDate.d == date.d);
```